Lib/test/test_configparser.py

```diff
@@ -980,6 +980,51 @@ def test_set_nonstring_types(self):
         self.assertRaises(TypeError, cf.set, "sect", 123, "invalid opt name!")
         self.assertRaises(TypeError, cf.add_section, 123)
 
+    def test_str(self):
+        self.maxDiff = None
+        cf = self.config_class(allow_no_value=True, delimiters=('=',), strict=True)
+        cf.add_section("sect1")
+        cf.add_section("sect2")
+        cf.set("sect1", "option1", "foo")
+        cf.set("sect2", "option2", "bar")
+
+        expected_str = (
+            "<ConfigParser: {'sect1': {'option1': 'foo'}, 'sect2': {'option2': 'bar'}}>"
+        )
+        self.assertEqual(str(cf), expected_str)
+
+    def test_repr(self):
+        self.maxDiff = None
+        cf = self.config_class(allow_no_value=True, delimiters=('=',), strict=True)
+        cf.add_section("sect1")
+        cf.add_section("sect2")
+        cf.add_section("sect3")
+        cf.add_section("sect4")
+        cf.add_section("sect5")
+        cf.add_section("sect6")
+        cf.set("sect1", "option1", "foo")
+        cf.set("sect2", "option2", "bar")
+        cf.read_string("")
+
+        dict_type = type(cf._dict).__name__
+        classname = cf.__class__.__name__
+        params = {
+            'dict_type': dict_type,
+            'allow_no_value': True,
+            'delimiters': ('=',),
+            'strict': True,
+            'default_section': 'DEFAULT',
+            'interpolation': 'BasicInterpolation',
+        }
+        state = {
+            'loaded_sources': ['<string>'],
+            'sections_count': 6,
+            'sections': ['sect1', 'sect2', 'sect3', 'sect4', 'sect5'],
+            'sections_truncated': '...and 1 more',
+        }
+        expected = f"<{classname}({params=}, {state=})>"
+        self.assertEqual(repr(cf), expected)
```

Comment:
```suggestion
        params = {
            'dict_type': type(cf._dict).__name__,
            'allow_no_value': True,
            'delimiters': ('=',),
            'strict': True,
            'default_section': 'DEFAULT',
            'interpolation': 'BasicInterpolation',
        }
        state = {
            'loaded_sources': ['<string>'],
            'sections_count': 6,
            'sections': ['sect1', 'sect2', 'sect3', 'sect4', 'sect5'],
            'sections_truncated': '...and 1 more',
        }
        expected = f"<{type(cf).__name__}({params=}, {state=})>"
        self.assertEqual(repr(cf), expected)
```

Sorry, I thought the `expected` wouldn't fit on 80 chars, but it fits, so it should be fine like this! By the way, I used `type(x).__name__` instead of `x.__class__.__name__` but I don't know which one is the best. I usually prefer the `__class__` one but I got told once that sometimes `type(x)` is preferrable but I can't remember why. So up to you (considering you used `type(cf._dict)`, I suggest using `type(cf)` as well).

Reply:
cpython codebase uses type(cf).__name__ I grepped one instance, __class__ can be overridden, but this is test , it's okay. 
btw, I guess except library it's okay, I saw the use of type(obj).__name__ in tools.